tests/events/events.go

```diff
@@ -0,0 +1,105 @@
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests/util"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type k8sObject interface {
+	metav1.Object
+	schema.ObjectKind
+}
+
+// ExpectNoEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectNoEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be not triggered")
+	expectEvent(object, eventType, reason, BeEmpty())
+}
+
+// ExpectEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be triggered")
+	expectEvent(object, eventType, reason, Not(BeEmpty()))
+}
+
+// DeleteEvents is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func DeleteEvents(object k8sObject, eventType, reason string) {
+	By("Expecting events to be removed")
+	virtClient, err := kubecli.GetKubevirtClient()
+	util.PanicOnError(err)
+
+	fieldSelector, namespace := constructFieldSelectorAndNamespace(object, eventType, reason)
+
+	events, err := virtClient.CoreV1().Events(namespace).List(
+		context.Background(),
+		metav1.ListOptions{
+			FieldSelector: fieldSelector,
+		},
+	)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	for _, event := range events.Items {
+		err = virtClient.CoreV1().Events(event.Namespace).Delete(context.Background(), event.Name, metav1.DeleteOptions{})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+	}
+
+	EventuallyWithOffset(1, func() []k8sv1.Event {
+		events, err := virtClient.CoreV1().Events(namespace).List(context.Background(),
+			metav1.ListOptions{
+				FieldSelector: fieldSelector,
+			})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+		return events.Items
+	}, 30*time.Second, 1*time.Second).Should(BeEmpty())
+}
+
+func expectEvent(object k8sObject, eventType, reason string, matcher types.GomegaMatcher) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(2, err).ToNot(HaveOccurred())
+
+	fieldSelector, namespace := constructFieldSelectorAndNamespace(object, eventType, reason)
+
+	EventuallyWithOffset(2, func() []k8sv1.Event {
+		events, err := virtClient.CoreV1().Events(namespace).List(
+			context.Background(),
+			metav1.ListOptions{
+				FieldSelector: fieldSelector,
+			},
+		)
+		ExpectWithOffset(3, err).ToNot(HaveOccurred())
+		return events.Items
+	}, 30*time.Second).Should(matcher)
+}
+
+func constructFieldSelectorAndNamespace(object k8sObject, eventType, reason string) (string, string) {
+	kind := object.GroupVersionKind().Kind
+	if kind == "" {
+		kind = reflect.ValueOf(object).Type().Name()
+	}
+	kindSelector := fmt.Sprintf("involvedObject.kind=%s,", kind)
+	if kind == "" {
```

Comment:
And which scenario is this one answers? Seems like it can happen when `reflect.ValueOf(object).Type().Name()` returns nothing. Can this even happen?

Reply:
Good question. I think I played it on the safe side here. But still, see
```
// Name returns the type's name within its package for a defined type.
// For other (non-defined) types it returns the empty string.
Name() [string](https://pkg.go.dev/builtin#string)
```

Reply:
Someone needs to do a lot of loops and hoops to pass in an object of an undefined type. Not even sure it is possible in this case.

But even if it is possible somehow, isn't this fitting an error? What will the `fieldSelector` result be if `kindSelector` is empty? Will it produce anything?